p2p/conn/connection.go

```diff
@@ -315,7 +337,18 @@ FOR_LOOP:
 			c.Logger.Debug("Send Ping")
 			wire.WriteByte(packetTypePing, c.bufWriter, &n, &err)
 			c.sendMonitor.Update(int(n))
+			c.Logger.Debug("Starting pong timer", "dur", c.config.PongTimeout)
+			c.pongTimer = time.AfterFunc(c.config.PongTimeout, func() {
```

Comment:
I think we need to stop the previous pongTimer, if it exists.

Reply:
1) there is no (little) chance it exists since above I check for `if config.PongTimeout >= config.PingInterval {`
2) that would be bad. what's the reason to start it in the first place if later we stop it and replace with newer timer

Reply:
seems correct but does Go guarantee it ? can we check if the pongTimer is still active ? if so, then logically it's timed out (unless we have a bug and fail to stop it when we receive the pong)